wallet-new/test/unit/Wallet/Abstract.hs

```diff
@@ -525,40 +541,35 @@ newtype InductiveGen h a
     { unInductiveGen :: ReaderT (InductiveCtx h) Gen a
     } deriving (Functor, Applicative, Monad, MonadReader (InductiveCtx h))
 
-runInductiveGen :: FromPreChain h () -> InductiveGen h a -> Gen a
+runInductiveGen :: FromPreTree h () -> InductiveGen h a -> Gen a
 runInductiveGen fpc ig = runReaderT (unInductiveGen ig) (initializeCtx fpc)
 
 data InductiveCtx h
     = InductiveCtx
-    { icFromPreChain  :: !(FromPreChain h ())
+    { icFromPreTree  :: !(FromPreTree h ())
     }
 
-initializeCtx :: FromPreChain h () -> InductiveCtx h
+initializeCtx :: FromPreTree h () -> InductiveCtx h
 initializeCtx fpc = InductiveCtx{..}
   where
-    icFromPreChain = fpc
-
-getFromPreChain :: InductiveGen h (FromPreChain h ())
-getFromPreChain = asks icFromPreChain
+    icFromPreTree = fpc
```

Comment:
Can be simplified to `initializeCtx icFromPreeTree = InductiveCtx {..}` or even `initializeCtx = InductiveCtx` though the latter will need to change more when the Ctx has more stuff